codewind-che-sidecar/deploy-pfe/pkg/codewind/deploy.go

```diff
@@ -14,20 +15,26 @@ import (
 
 // DeployCodewind takes in a `codewind` object and deploys Codewind and the performance dashboard into the specified namespace
 func DeployCodewind(clientset *kubernetes.Clientset, codewind Codewind, namespace string) error {
-	// Create a PVC for PFE
-	// Determine if we're running on OpenShift on IKS (and thus need to use the ibm-file-bronze storage class)
-	storageClass := ""
-	sc, err := clientset.StorageV1().StorageClasses().Get(constants.ROKSStorageClass, metav1.GetOptions{})
-	if err == nil && sc != nil {
-		storageClass = sc.Name
-		log.Infof("Setting storage class to %s\n", storageClass)
-	}
-
-	pvc := generatePVC(codewind, constants.PFEVolumeSize, storageClass)
-	_, err = clientset.CoreV1().PersistentVolumeClaims(namespace).Create(&pvc)
+	// See if a PVC for the PFE workspace already exists, if not, create one
+	_, err := clientset.CoreV1().PersistentVolumeClaims(namespace).Get(codewind.PVCName, metav1.GetOptions{})
 	if err != nil {
-		log.Errorf("Unable to create Persistent Volume Claim for PFE: %v\n", err)
-		return err
+		// Create a PVC for PFE
+		// Determine if we're running on OpenShift on IKS (and thus need to use the ibm-file-bronze storage class)
+		storageClass := ""
+		sc, err := clientset.StorageV1().StorageClasses().Get(constants.ROKSStorageClass, metav1.GetOptions{})
+		if err == nil && sc != nil {
+			storageClass = sc.Name
+			log.Infof("Setting storage class to %s\n", storageClass)
+		}
+
+		// Get the name and uid for the Che workspace volume
+		chePvc := che.GetWorkspacePVC(clientset, namespace, codewind.WorkspaceID)
+		pvc := generatePVC(codewind, constants.PFEVolumeSize, storageClass, chePvc.GetObjectMeta().GetName(), chePvc.GetObjectMeta().GetUID())
```

Comment:
Should we move the call to `che.GetWorkspacePVC()` into the`generatePVC` function instead and only use its name and uid if there is a workspace pvc and default to the codewind ReplicaSet name and uid if it's not available? We'd need to update `che.GetWorkspacePVC()` to return an error instead of exit as well but that way it'd work for hybrid scenarios too where there wouldn't be a che PVC.

Reply:
I'd say leave as-is for now. When we go to consolidate the Che and Hybrid deployment code, I want to change how we're detecting we're running in Che, and don't want a check for Che in `generatePVC` either.